### PR TITLE
Allow middle-click to close scripts

### LINKED ISSUE #20871: Closing script files with middle mouse button click
We can close scene tabs with middle-click. This helps to quickly clean up a bunch of unused tabs.

The same should be supported for closing script files in the script editor. Right now, middle-click on a file name does nothing.

I'm going to take a look to see if I can make this happen quickly. If not, I'll leave this issue open for someone else to pick up.

### PR DESCRIPTION
**Pull request by ttencate**
Fixes #20871 

My first Godot PR, it's not much but it's something :) I just put together some existing pieces, so I hope this isn't too horribly borken.

### CODE REVIEW

File: editor/plugins/script_editor_plugin.cpp [hunk 1]
```diff
@@ -2384,9 +2384,23 @@ void ScriptEditor::_unhandled_input(const Ref<InputEvent> &p_event) {
 void ScriptEditor::_script_list_gui_input(const Ref<InputEvent> &ev) {
 
 	Ref<InputEventMouseButton> mb = ev;
-	if (mb.is_valid() && mb->get_button_index() == BUTTON_RIGHT && mb->is_pressed()) {
+	if (mb.is_valid() && mb->is_pressed()) {
+		switch (mb->get_button_index()) {
+
+			case BUTTON_MIDDLE: {
+				// Right-click selects automatically; middle-click does not.
+				int idx = script_list->get_item_at_position(mb->get_position(), true /* p_exact */);
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.cpp]: Please don't put the argument name as inline comment, it's the job of your IDE to tell you what the method signature is if not clear.

Comment (ttencate) [reply editor/plugins/script_editor_plugin.cpp]: Thanks for the review! Google style guide ruined me. Fix pushed.

**Comment by akien-mga** [comment]
Could you squash the commits?

**Comment by ttencate** [comment]
Done! Thanks for your patience :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**